src/pages/ni/planning/AuxiliaryPlanning.vue

```diff
@@ -243,44 +265,52 @@ export default {
       };
       this.creationModal = true;
     },
+    formatAuxiliaryWithSector (aux, sectorHistory) {
+      return { ...aux, sector: sectorHistory ? sectorHistory.sector : null };
+    },
+    getSectorHistory (aux, sector) {
+      const sectorHistory = aux.sectorHistories.find(sectorHistory => {
+        const isSameSector = sector ? sectorHistory.sector._id === sector._id : true;
+        const isStartDateBeforeEndOfWeek = this.$moment(sectorHistory.startDate).isSameOrBefore(this.endOfWeek);
+        const isEndDateAfterStartOfWeek = this.$moment(sectorHistory.endDate).isSameOrAfter(this.startOfWeek);
+        return isSameSector && isStartDateBeforeEndOfWeek && isEndDateAfterStartOfWeek;
+      });
+      return sectorHistory;
+    },
+    getAuxBySector (sector) {
+      return this.filters.filter(aux => {
+        if (!aux.sectorHistories) return false;
+        return this.getSectorHistory(aux, sector);
```

Comment:
tu peux mettre 
```
return this.filters.filter(aux => !!aux.sectorHistories && !!this.getSectorHistory(aux, sector)); 
```
non ?

Reply:
ou alors le check sur `aux.sectorHistories` tu le passes dans `getSectorHistory `
(`if (!aux.sectorHIstories) return null`)
et la tu mets juste 
`return this.filters.filter(aux => !!this.getSectorHistory(aux, sector)); `